Yank/multistate/sams.py

```diff
@@ -0,0 +1,419 @@
+    >>> temperatures = [T_min + (T_max - T_min) * (math.exp(float(i) / float(nreplicas-1)) - 1.0) / (math.e - 1.0)
+    ...                 for i in range(n_replicas)]
+    >>> thermodynamic_states = [states.ThermodynamicState(system=testsystem.system, temperature=T)
+    ...                         for T in temperatures]
+
+    Initialize simulation object with options. Run with a GHMC integrator:
+
+    >>> move = mcmc.GHMCMove(timestep=2.0*unit.femtoseconds, n_steps=50)
+    >>> simulation = SAMSSampler(mcmc_moves=move, number_of_iterations=2,
+    >>>                          state_update_scheme='restricted-range', locality=5,
+    >>>                          update_stages='two-stage', flatness_threshold=0.2,
+    >>>                          weight_update_method='rao-blackwellized',
+    >>>                          adapt_target_probabilities=False)
+
+
+    Create a single-replica SAMS simulation bound to a storage file and run:
+
+    >>> storage_path = tempfile.NamedTemporaryFile(delete=False).name + '.nc'
+    >>> reporter = MultiStateReporter(storage_path, checkpoint_interval=1)
+    >>> simulation.create(thermodynamic_states=thermodynamic_states,
+    >>>                   sampler_states=states.SamplerState(testsystem.positions),
+    >>>                   storage=reporter)
+    >>> simulation.run()  # This runs for a maximum of 2 iterations.
+    >>> simulation.iteration
+    2
+    >>> simulation.run(n_iterations=1)
+    >>> simulation.iteration
+    2
+
+    To resume a simulation from an existing storage file and extend it beyond
+    the original number of iterations.
+
+    >>> del simulation
+    >>> simulation = ReplicaExchangeSampler.from_storage(reporter)
+    >>> simulation.extend(n_iterations=1)
+    >>> simulation.iteration
+    3
+
+    You can extract several information from the NetCDF file using the Reporter
+    class while the simulation is running. This reads the SamplerStates of every
+    run iteration.
+
+    >>> reporter = MultiStateReporter(storage=storage_path, open_mode='r', checkpoint_interval=1)
+    >>> sampler_states = reporter.read_sampler_states(iteration=range(1, 4))
+    >>> len(sampler_states)
+    3
+    >>> sampler_states[-1].positions.shape  # Alanine dipeptide has 22 atoms.
+    (22, 3)
+
+    Clean up.
+
+    >>> os.remove(storage_path)
+
+    See Also
+    --------
+    ReplicaExchangeSampler
+
+    """
+
+    _TITLE_TEMPLATE = ('Self-adjusted mixture sampling (SAMS) simultion using SAMSSampler '
+                       'class of yank.multistate on {}')
+
+    def __init__(self,
+                log_target_probabilities=None,
+                state_update_scheme='restricted-range', locality=5,
+                update_stages='two-stage', flatness_threshold=0.2,
+                weight_update_method='rao-blackwellized',
+                adapt_target_probabilities=False,
+                **kwargs):
+        """Initialize a SAMS sampler.
+
+        Parameters
```

Comment:
I think you can define the parameters for class `__init__` in the main docstring, that way there is only one place people need to look and you have to maintain